File: app/src/main/java/fr/skyost/timetable/services/TodayWidgetService.java
package fr.skyost.timetable.services;

import android.content.Intent;
import android.widget.RemoteViewsService;

import fr.skyost.timetable.receivers.TodayWidgetReceiver;

public class TodayWidgetService extends RemoteViewsService {

	@Override
	public final RemoteViewsFactory onGetViewFactory(final Intent intent) {
		return new TodayWidgetViewsFactory(this);
	}

}

File: android/app/src/main/kotlin/fr/skyost/timetable/widget/TodayWidgetService.kt
package fr.skyost.timetable.widget

import android.appwidget.AppWidgetManager
import android.content.Intent
import android.widget.RemoteViewsService

/**
 * The today's widget service.
 */
class TodayWidgetService : RemoteViewsService() {
    override fun onGetViewFactory(intent: Intent): RemoteViewsFactory {
        // We create the corresponding RemoteViews factory.
        val widgetId = intent.getIntExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, AppWidgetManager.INVALID_APPWIDGET_ID)
        return TodayWidgetViewsFactory(applicationContext, widgetId)
    }
}
